### ISSUE TITLE
conv2d overloads ambiguous because of `stride` parameter

### ISSUE THREAD

**Issue by dsyme** [DiffSharp/DiffSharp#99]
These overloads are ambiguous unless an explicit `stride` is given:
```
    static member conv2d(a:Tensor, b:Tensor, ?stride:seq<int>, ?padding:seq<int>, ?dilation:seq<int>) = a.conv2d(b, ?stride=stride, ?padding=padding, ?dilation=dilation)
    static member conv2d(b:Tensor, ?stride:seq<int>, ?padding:seq<int>, ?dilation:seq<int>) = fun (a:Tensor) -> a.conv2d(b, ?stride=stride, ?padding=padding, ?dilation=dilation)
    static member conv2d(a:Tensor, b:Tensor, ?stride:int, ?padding:int, ?dilation:int) = a.conv2d(b, ?stride=stride, ?padding=padding, ?dilation=dilation)
    static member conv2d(b:Tensor, ?stride:int, ?padding:int, ?dilation:int) = fun (a:Tensor) -> a.conv2d(b, ?stride=stride, ?padding=padding, ?dilation=dilation)
```
e.g.
```
            let x = dsharp.ones([1;1;4;4])
            let y = dsharp.ones([1;1;4;4])
            let z = dsharp.conv2d(x,y)
```
gives
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	FS0041	A unique overload for method 'conv2d' could not be determined based on type information prior to this program point. A type annotation may be needed. Candidates: static member DiffSharp.conv2d : a:Tensor * b:Tensor * ?stride:int * ?padding:int * ?dilation:int -> Tensor, static member DiffSharp.conv2d : a:Tensor * b:Tensor * ?stride:seq<int> * ?padding:seq<int> * ?dilation:seq<int> -> Tensor	DiffSharp.Tests	C:\GitHub\dsyme\DiffSharp\src\DiffSharp.Tests\TestTensor.fs	1957	Active
```

**Comment by dsyme** [comment]
A solution is to have two optional parameters `stride` and `strides` and ensure only one is given

**Comment by kevmal** [comment]
Looks like any one of `stride`, `padding`, or `dilation` could be given. So I guess you'd need a plural version for each... Which would be useful since currently one parameter forces the other parameters to be consistent (all ints or all int seqs where with PyTorch it's common to mix and match). Alternatively maybe just drop the `int` overloads and allow seqs of length 1.

**Comment by gbaydin** [comment]
Hi, yes, I converged to the same solution independently in the convnet work that is currently in a separate branch. :)

https://github.com/DiffSharp/DiffSharp/blob/feature/convnets/src/DiffSharp.Core/DiffSharp.fs#L158

**Comment by gbaydin** [comment]
Closing this as the fix was merged in #96
